hangman.py

```diff
@@ -29,20 +39,155 @@
+    is_exiting(word)
+    return word
+  elif ai:
+    is_exiting(word_choice)
+    if word_choice == "easy":
+      return random.choice(easy)
+    if word_choice == "medium":
+      return random.choice(medium)
+    if word_choice == "hard":
+      return random.choice(hard)
+    if word_choice == "impossible":
+      return random.choice(impossible)
+    if word_choice == "any":
+      return random.choice(any)
+def get_guess(guessed ,ai=False):
+  # This gets the guess from the second player or the potential AI
+  if not ai:
+    good_guess = False
+    while not good_guess:
+      guess = input("What is your guess? (1 letter)")
+      
+      is_exiting(guess)
+      if guess == "~":
+        return "~"
+      if guess not in guessed and len(guess) == 1:
+        return guess
+def win(answer, current):
+  # check if the player's guesses are the word
+  if answer == current:
+    return True
+  return False
+def print_gallows(lives):
+  # This is to get the proper ascii art for the gallows
+  print(HANGMAN_PICS[lives])
+def is_dead(lives):
+  # check if the player is dead
+  if lives <= 0:
+    return True
+  return False
 
 # `play_hangman` is the main function, the function
 # that will orchestrate all the helper functions
 # you define, above.
 def play_hangman():
-    pass
-
+    # Get secret word and listify
+    got_is_ai = False
+    while not got_is_ai:
+      is_ai = input("singleplayer?? y/n: ")
+      is_exiting(is_ai)
+      if is_ai == "y":
+        got_difficulty = False
+        while not got_difficulty:
+          difficulty_set = input("difficulty? easy, medium, hard, impossible, or any: ")
+          is_exiting(difficulty_set)
+          if difficulty_set in ai_settings:
+            got_difficulty = True
+          print("Spell correctly you idiot")
+        secret_word = (get_word(True, difficulty_set)).lower()
+        got_is_ai = True
+      if is_ai == "n":
+        secret_word = (get_word()).lower()
+        got_is_ai = True
+      else:
+        print('please do not be the gay retard you are and just say "y" or "n".')
+    sw_list = []
+    # make displayed list
+    disp_word = []
+    # whether the word is guessed or lives run out
+    guessed = False
+    lives = 6
+    # Make the lists with proper length
+    for l in secret_word:
+      sw_list.append(l)
+      disp_word.append("_")
+    clear()
+    # print(sw_list) # Test
+    # list of guessed letters
+    guessed_letters = []
+    # repeat until the word is guessed
+    while not guessed:
+      print(disp_word)
+      print("you already guessed:", guessed_letters)
+      guess = (get_guess(guessed_letters)).lower()
+      if guess == "~":
+        clear()
+        print("You Win!!!")
+        print("""
+\      /\      /  
+ \    /  \    /
+  \  /    \  /
+   \/      \/
+""")
+        print(sw_list)
+        return "Success"
+      clear()
+      # Check to see if the guess is correct
+      got_letter = False
+      indexed = 0
+      for g in sw_list:
```

Comment:
A more "Pythonic" way to get the index along with the value:
```
for index, g in enumerate(sw_list):
```

`enumerate()` creates a generator that will return, for each element in `sw_list`, a tuple: `(<index>, <value>)`. Then you don't have to initialize and increment and counter variable.